src/include/platform/NetworkCommissioning.h

```diff
@@ -350,6 +351,14 @@ class WiFiDriver : public Internal::WirelessDriver
     }
 #endif // CHIP_DEVICE_CONFIG_ENABLE_WIFI_PDC
 
+    /**
+     *  @brief Provide all the frequency bands supported by the Wi-Fi interface
+     *
+     *  Provide a default implementation that returns the 2.4 Ghz band support.
+     *  Note: WiFi platforms should implement this function in their WiFiDriver to provide their complete device capabilities
+     */
+    virtual WiFiBand GetSupportedWiFiBands() { return WiFiBand::k2g4; }
```

Comment:
Why are we defaulting this at all, instead of forcing it to be implemented, like the Thread callbacks?

Reply:
There is a total of 17 different WiFiDriver implementations. I opted to provide a default basic implementation rather than coping the same one for all 17. I expect the 17 platform implementation will want to override this function and fetch their real/complete band support. Since they all have different API/config to get that information I want to leave that to each platform dev.

If preferred, I can copy this default one to the 17 implementation and let each platform dev rewrite it when they see fit

Reply:
Probably fine as is, but maybe file an issue on making sure people do it and then removing the fallback?

Reply:
Will open the issue

Reply:
Issue created
https://github.com/project-chip/connectedhomeip/issues/30579